contracts/src/token/common/erc2981.rs

```diff
@@ -0,0 +1,514 @@
+#[storage]
+#[derive(Erase)]
+pub struct RoyaltyInfo {
+    receiver: StorageAddress,
+    royalty_fraction: StorageU96,
+}
+
+/// State of an ERC2981 contract.
+#[storage]
+pub struct Erc2981 {
+    /// The default royalty information for all tokens.
+    pub(crate) default_royalty_info: RoyaltyInfo,
+    /// The royalty information for a particular token.
+    /// It is a mapping of `token_id` to `RoyaltyInfo`.
+    pub(crate) token_royalty_info: StorageMap<U256, RoyaltyInfo>,
+}
+
+/// Interface for the NFT Royalty Standard.
+
+/// A standardized way to retrieve royalty payment information for non-fungible
+/// tokens (NFTs) to enable universal support for royalty payments across all
+/// NFT marketplaces and ecosystem participants.
+#[interface_id]
+pub trait IErc2981 {
+    /// Returns how much royalty is owed and to whom, based on a sale price that
+    /// may be denominated in any unit of exchange.
+    ///
+    /// The royalty amount is denominated and should be paid in that same unit
+    /// of exchange.
+    ///
+    /// NOTE: ERC-2981 allows setting the royalty to 100% of the price.
+    /// In that case all the price would be sent to the royalty receiver and 0
+    /// tokens to the seller. Contracts dealing with royalty should consider
+    /// empty transfers.
+    ///
+    /// # Arguments
+    ///
+    /// * `&self` - Write access to the contract's state
+    /// * `token_id` - Token id of the token
+    /// * `sale_price` - The sale price of the token
+    fn royalty_info(&self, token_id: U256, sale_price: U256)
+        -> (Address, U256);
+}
+
+/// NOTE: Implementation of [`TopLevelStorage`] to be able use `&mut self` when
+/// calling other contracts and not `&mut (impl TopLevelStorage +
+/// BorrowMut<Self>)`. Should be fixed in the future by the Stylus team.
+unsafe impl TopLevelStorage for Erc2981 {}
+
+#[public]
+impl IErc2981 for Erc2981 {
+    fn royalty_info(
+        &self,
+        token_id: U256,
+        sale_price: U256,
+    ) -> (Address, U256) {
+        let royalty_info = self.token_royalty_info.get(token_id);
+        let mut royalty_receiver = &royalty_info.receiver;
+        let mut royalty_fraction = &royalty_info.royalty_fraction;
+
+        if royalty_receiver.is_zero() {
+            royalty_receiver = &self.default_royalty_info.receiver;
+            royalty_fraction = &self.default_royalty_info.royalty_fraction;
+        }
+
+        let royalty_amount = (sale_price * U256::from(royalty_fraction.get()))
+            .wrapping_div(U256::from(self._fee_denominator()));
+
+        (royalty_receiver.get(), royalty_amount)
+    }
+}
+
+impl IErc165 for Erc2981 {
+    fn supports_interface(interface_id: FixedBytes<4>) -> bool {
+        <Self as IErc2981>::INTERFACE_ID == u32::from_be_bytes(*interface_id)
+    }
+}
+
+impl Erc2981 {
+    /// Function to change the denominator with which to interpret the fee set
+    /// in _setTokenRoyalty and _setDefaultRoyalty as a fraction of the sale
```

Comment:
```suggestion
    /// Changes the denominator with which to interpret the fee set
    /// in _setTokenRoyalty and _setDefaultRoyalty as a fraction of the sale
```
nit: it's obviously a function. Update all relevant comments the same way.
Also this should link to the actual Rust functions